influxdb/client.go

```diff
@@ -281,19 +309,93 @@ func (c *HTTPClient) do(req *http.Request, result interface{}, codes ...int) (*h
 		if rp.Error != "" {
 			return nil, errors.New(rp.Error)
 		}
-		return nil, fmt.Errorf("invalid response: code %d: body: %s", resp.StatusCode, string(body))
+		return nil, fmt.Errorf("invalid response: code %d", resp.StatusCode)
 	}
+
 	if result != nil {
-		d := json.NewDecoder(resp.Body)
+		d := json.NewDecoder(body)
 		d.UseNumber()
-		err := d.Decode(result)
+		err = d.Decode(result)
 		if err != nil {
 			return nil, errors.Wrap(err, "failed to decode JSON")
 		}
 	}
 	return resp, nil
 }
 
+func (c *HTTPClient) doFlux(req *http.Request, codes ...int) (*Response, error) {
```

Comment:
Since this is not implemented anyway - how about having this return a ReadCloser and we can have a QueryFluxResponse to parse and return a `Response`, and a QueryFlux to return a `ResultIterator` ?

We can do this post-merge, wondering what your preference on API is.

Reply:
I like this idea.